### PR TITLE
chore: fix dag issues

### PR DESCRIPTION
**Pull request by v3g42**

### CODE REVIEW

File: dozer-core/src/dag/executor.rs [hunk 1]
```diff
@@ -166,43 +167,75 @@ impl<'a, T: Clone + Debug + 'static> DagExecutor<'a, T> {
...
+            return Err(IncompatibleSchemas(
+                "Input Schemas length mismatch".to_string(),
+            ));
         }
-        for (port, (schema, _ctx)) in &current.output_schemas {
-            let other_schema = existing
-                .output_schemas
-                .get(port)
-                .ok_or(IncompatibleSchemas())?;
+        for (port, (schema, _ctx)) in &current.input_schemas {
```
Comment (chloeminkyung) [inline dozer-core/src/dag/executor.rs:217]: should be `output_schemas`?

Comment (chubei) [reply dozer-core/src/dag/executor.rs:217]: I'm not sure. @v3g42 said it was a fix.

Comment (v3g42) [reply dozer-core/src/dag/executor.rs:217]: No. I am not sure about this code actually. I saw two code blocks that repeated. I assumed one was for input_schemas. Can you check the logic properly again ?

Comment (chubei) [reply dozer-core/src/dag/executor.rs:217]: This commit's version is correct. Resovling this conversation.

**Review by chloeminkyung — APPROVED** [review]

**Pull request merged**